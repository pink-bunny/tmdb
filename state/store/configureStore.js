import { createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';

import reducer from './reducer';

const loggerMiddleware = (store) => (next) => (action) => {
  /* eslint-disable no-console */
  console.group(action.type);
  console.info('dispatching', action);
  const result = next(action);
  console.log('next state', store.getState());
  console.groupEnd();
  /* eslint-enable */

  return result;
};

export default function configureStore() {
  const middlewares = [loggerMiddleware, thunkMiddleware];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const store = createStore(reducer, undefined, composeWithDevTools(middlewareEnhancer));

  return store;
}
