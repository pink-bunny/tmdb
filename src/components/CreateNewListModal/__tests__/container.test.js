import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';

import ConnectedCreateNewListModal, { handleSubmit } from '../container';

describe('CreateNewListModal', () => {
  const store = configureStore()({});
  const requiredProps = {
    store,
    modalVisible: false,
    showModal: jest.fn(),
    triggerComponent: jest.fn(() => <div>Mock button</div>)
  };
  store.dispatch = jest.fn();

  describe('container', () => {
    describe('without trigger props', () => {
      const wrapper = shallow(<ConnectedCreateNewListModal {...requiredProps} />); /* eslint-disable-line react/jsx-props-no-spreading, max-len */
      const container = wrapper.dive().dive().dive().dive();
      const instance = container.instance();
      const { showModal } = requiredProps;

      it('matches snapshot', () => {
        expect(container).toMatchSnapshot();
      });

      it('handleTriggerClick() called only showModal()', () => {
        instance.handleTriggerClick();
        expect(showModal).toHaveBeenCalled();
      });
    });

    describe('with trigger props', () => {
      const props = {
        ...requiredProps,
        triggerProps: {
          classname: 'mock-button',
          onClick: jest.fn()
        }
      };
      const wrapper = shallow(<ConnectedCreateNewListModal {...props} />); /* eslint-disable-line react/jsx-props-no-spreading, max-len */
      const container = wrapper.dive().dive().dive().dive();
      const instance = container.instance();
      const { triggerProps, showModal } = props;

      it('matches snapshot', () => {
        expect(container).toMatchSnapshot();
      });

      it('handleTriggerClick() called triggerProps.onClick() and showModal()', () => {
        instance.handleTriggerClick();
        expect(triggerProps.onClick).toHaveBeenCalled();
        expect(showModal).toHaveBeenCalled();
      });
    });
  });

  describe('form.', () => {
    describe('handleSubmit()', () => {
      const props = {
        hideModal: jest.fn(),
        handleSubmitList: jest.fn()
      };
      const setErrors = jest.fn();
      const values = {};

      it('calls createMyList with arguments', () => {
        handleSubmit(values, { props, setErrors });
        expect(props.handleSubmitList).toHaveBeenCalledWith(values, setErrors, props.hideModal);
      });
    });

    describe('validationSchema', () => {
      const wrapper = shallow(<ConnectedCreateNewListModal {...requiredProps} />); /* eslint-disable-line react/jsx-props-no-spreading, max-len */
      const container = wrapper.dive().props().validationSchema()
        .describe();

      it('matches snapshot', () => {
        expect(container).toMatchSnapshot();
      });
    });
  });
});
