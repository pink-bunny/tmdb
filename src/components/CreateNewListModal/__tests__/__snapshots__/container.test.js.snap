// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateNewListModal container with trigger props matches snapshot 1`] = `
<CreateNewListModal
  errors={Object {}}
  handleSubmit={[Function]}
  handleTriggerClick={[Function]}
  hideModal={[Function]}
  modalVisible={false}
  showModal={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  triggerComponent={[MockFunction]}
  triggerProps={
    Object {
      "classname": "mock-button",
      "onClick": [MockFunction],
    }
  }
/>
`;

exports[`CreateNewListModal container without trigger props matches snapshot 1`] = `
<CreateNewListModal
  errors={Object {}}
  handleSubmit={[Function]}
  handleTriggerClick={[Function]}
  hideModal={[Function]}
  modalVisible={false}
  showModal={[MockFunction]}
  triggerComponent={[MockFunction]}
  triggerProps={null}
/>
`;

exports[`CreateNewListModal form. validationSchema matches snapshot 1`] = `
Object {
  "fields": Object {
    "description": Object {
      "label": undefined,
      "meta": undefined,
      "tests": Array [
        Object {
          "name": "required",
          "params": undefined,
        },
      ],
      "type": "string",
    },
    "name": Object {
      "label": undefined,
      "meta": undefined,
      "tests": Array [
        Object {
          "name": "required",
          "params": undefined,
        },
      ],
      "type": "string",
    },
  },
  "label": undefined,
  "meta": undefined,
  "tests": Array [],
  "type": "object",
}
`;
